mindsdb/integrations/utilities/time_series_utils.py

```diff
@@ -65,10 +66,10 @@ def get_results_from_nixtla_df(nixtla_df, model_args):
 
 def infer_frequency(df, time_column, default=DEFAULT_FREQUENCY):
     try:  # infer frequency from time column
-        date_series = pd.to_datetime(df[time_column]).unique()
-        if isinstance(date_series, np.ndarray):
-            date_series.sort()
-        inferred_freq = pd.infer_freq(date_series)
+        date_series = pd.to_datetime(df.sort_values(by=time_column)[time_column]).unique()
```

Comment:
Obs: this is not robust. E.g. given a monthly dataset, it will yield a specific time distance (most common), `nD` instead of the correct `1M`. We should improve this, @pedrofluxa any ideas? If not, we can try later and unblock the microsecond use case for now with what we have.

Reply:
Solution: first try `pd.infer_freq`, and if it does not work follow up with this approach.